### PR TITLE
Demographics & decentralized countries

### PR DESCRIPTION
**Pull request by Zemurin**
- previously imported EU4 demographics, being copied from chunks into substates, are now converted into Vic3 demographics.
- this allows for iteration over substates in order to coalesce them into new decentralized countries, filtered by primary culture.

### CODE REVIEW

File: EU4toV3/Source/Mappers/CountryMapper/CountryMapper.cpp [hunk 1]
```diff
@@ -187,3 +187,10 @@ std::string mappers::CountryMapper::mapToTag(const std::string& eu4Tag, const st
 
 	return newTag;
 }
+
+std::string mappers::CountryMapper::requestNewV3Tag()
+{
+	auto newTag = generateNewTag();
+	unmappedV3Tags.emplace(newTag);
+	return newTag;
+}
```
Comment (Idhrendur) [inline EU4toV3/Source/Mappers/CountryMapper/CountryMapper.cpp:196]: Tests

File: EU4toV3/Source/V3World/ClayManager/StateLoader/State.h [hunk 2]
```diff
@@ -15,7 +15,7 @@ class State: commonItems::parser
 	void loadState(std::istream& theStream);
 	void setStateName(const std::string& theName) { name = theName; }
 	void addSubState(const std::shared_ptr<SubState>& substate) { substates.push_back(substate); }
-	void distributeLandshares();
+	void distributeLandshares() const;
```
Comment (Idhrendur) [inline EU4toV3/Source/V3World/ClayManager/StateLoader/State.h:18]: const is technically correct, but misleading here. You aren't changing the pointers, but you are changing the underlying objects.

File: EU4toV3/Source/V3World/PoliticalManager/PoliticalManager.cpp [hunk 3]
```diff
@@ -47,3 +49,83 @@ void V3::PoliticalManager::importVanillaCountries()
...
+		else
+			cultureScores.emplace(demo.culture, demo.upperRatio);
+	}
+
+	const auto highest = std::max_element(std::begin(cultureScores), std::end(cultureScores), [](const auto& p1, const auto& p2) {
+		return p1.second < p2.second;
+	});
+
+	return highest->first;
+}
```
Comment (Idhrendur) [inline EU4toV3/Source/V3World/PoliticalManager/PoliticalManager.cpp:131]: I find the namespace {} trick I use to be a bit more readable than const/static private member functions. No need to even have them in the class declaration at all.

**Review by Idhrendur — APPROVED** [review]

File: EU4toV3/Source/V3World/ClayManager/StateLoader/State.h (hunk 2 above)
Comment (Zemurin) [reply EU4toV3/Source/V3World/ClayManager/StateLoader/State.h:18]: I'm aware, but that's the only way to appease resharper.

File: EU4toV3/Source/V3World/PoliticalManager/PoliticalManager.cpp (hunk 3 above)
Comment (Zemurin) [reply EU4toV3/Source/V3World/PoliticalManager/PoliticalManager.cpp:131]: Is it though? This way I can take a quick glance at header file and see what functions are available. You don't list those nameless functions in a header so having them become loose is quite a possibility. Also there's the resharper support for "hey you don't use this method anywhere, can delete it."

**Pull request merged**